plugins/parodos/src/components/workflow/workflowDetail/WorkFlowDetail.tsx

```diff
@@ -34,25 +38,33 @@ export const WorkFlowDetail = () => {
 
   // update task state regularly
   useEffect(() => {
-    const updateWorkflowExecutionState = async (): Promise<WorkflowTask[]> => {
+    const updateWorks = (works: WorkStatus[]) => {
+      let needUpdate = false;
+      works?.forEach(work => {
```

Comment:
I'm not saying you have to change this and it certainly is not wrong but I favour `for .. of ..` and I think a lot do 

e.g. 

```ts
    // initialise empty arrays where possible
    const updateWorks = (works: WorkStatus[] = []) => {
      let needUpdate = false;
      
      for (const work of works ) {
```

You can get into some problems modifying things with and you cannot `break` or `continue`.

[Some context](https://medium.com/front-end-weekly/3-things-you-didnt-know-about-the-foreach-loop-in-js-ff02cec465b1).

Reply:
totally agree. I need to follow best practice